lib/src/infrasructure/repositry/get_servieces_repositry/get_servieces_repositry.dart

```diff
@@ -11,45 +12,33 @@ import 'package:shared_preferences/shared_preferences.dart';
 import 'package:coofix/src/presentation/core/constants/constants.dart';
 
 /// TODO : Review changes to do
-/// 
+///
 /// - Make api calling and its related functionalities such as interception, error handling as a seperate service class
 /// - Add proper status code handling method. Try to use the features of DioIntercepter for it.
 /// - User debugPrint or log instead of print statement.
 /// - Craete seperate api service profiles for general and profile apis.
-/// - Handle Heeders with tokens for different api profiles commonly 
+/// - Handle Heeders with tokens for different api profiles commonly
 /// - Find a common method for api calls to reduce reduntant writing of same code in every repositories.
 /// - Remove unused log, print statements
 
-@LazySingleton(as: IGetServieces)
-class GetServiecesRepositry implements IGetServieces {
+@LazySingleton(as: IGetServiecesRepository)
+class GetServiecesRepository implements IGetServiecesRepository {
+  final Api api;
+  GetServiecesRepository({required this.api});
   @override
   Future<List<GetServiecesModel>> getServices(
       {required String id, required int skip, required int limit}) async {
     try {
-      SharedPreferences prefs = await SharedPreferences.getInstance();
       final Map<String, dynamic> data = {
         'id': id,
         'limit': limit,
         'skip': skip,
       };
-      var dio = Dio();
-      dio.interceptors.add(Dionterceptor());
-      final response = await dio.post(
-        ApiEndpoints.getServieces,
-        data: data,
-        options: Options(headers: {"Tokenvalid": AppConstants.tockenValied}),
-      );
-      if (kDebugMode) {
-        print(
-            "get services accsess token is ${prefs.getString('access_token')}");
-        print('Get Servieces Response status code: ${response.statusCode}');
-        print('Get Servieces Response headers: ${response.headers}');
-        print('Get Servieces Response body: ${response.data}');
-      }
+
+      var response =
+          await api.profile.post(ApiEndpoints.getServieces, data: data);
       if (response.data is Map<String, dynamic>) {
```

Comment:
No need to check the response type is Map or not it will be catched automatically. Remove it